decode/performance.ts

```diff
@@ -108,13 +108,17 @@ export function decode(tokens: Token[]): PerformanceEvent  {
 }
 
 function process(network: any[] | number[], stringIndex: number): NetworkData {
+    // Checking for null values since different browsers implement
+    // resource timing spec differently. For instance, not all version of Edge support
+    // transferSize property.
+    let empty = "";
     return {
-        start: network[0] as number,
-        duration: network[1] as number,
-        size: network[2] as number,
+        start: network[0] === -1 ? null : network[0] as number,
+        duration: network[1] === -1 ? null : network[1] as number,
+        size: network[2] === -1 ? null : network[2] as number,
         target: stringIndex > 3 ? network[3] as number : null,
-        initiator: network[stringIndex] as string,
-        protocol: network[stringIndex + 1] as string,
-        host: network[stringIndex + 2] as string
```

Comment:
Please fix the typing declaration ni thing like NetworkData to indicate that the field is nullable. For example it's like this now:

export interface NetworkData {
    start: number;
    duration: number;
    size: number;
    target: Target;
    initiator: string;
    protocol: string;
    host: string;
}

I think it should be something like: 

export interface NetworkData {
    start: number | null;
    duration: number | null;
    size: number | null
    target: Target;
    initiator?: string;
    protocol?: string;
    host?: string;
}

One other nuance is the difference between attribute not present vs. attribute is present and null. See above where I wasn't sure in some cases. Please check other cases where you have this kind if "null handling" for variably-supported features. 

And that's a best guess It would be very helpful to get this precise as it's what I use to define the matching schema on the pipeline were what is optional, or nullable (not same) needs to be exactly in sync with your code.

Reply:
I've simplified this. (?) in typescript means the field is optional, which these are not. However if the value is missing, it will acquire a null value - which is a valid type: number or string. This is consistent with other places in the code. Take a look with refactored code and let me know your thoughts. Resolving this for now. 

---
In reply to: [369057739](https://github.com/microsoft/clarity-js/pull/166#discussion_r369057739) [](ancestors = 369057739)